File: SWEA/D4/1233. ［S／W 문제해결 기본］ 9일차 － 사칙연산 유효성 검사/［S／W 문제해결 기본］ 9일차 － 사칙연산 유효성 검사.py
opp = ['+', '-', '*', '/']
for testcase in range(1, 11):
    N = int(input())
    flag = True
    for _ in range(N):
        query = list(input().rstrip().split())
        if len(query) == 2:
            node, value = query
            if value in opp:
                flag = False
        elif len(query) >= 3:
            _, a, b = query[0], query[1], query[:]
            if a not in opp:
                flag = False
                
    print("#{} {}".format(testcase, int(flag)))

File: SWEA/D4/1233. ［S／W 문제해결 기본］ 9일차 － 사칙연산 유효성 검사/［S／W 문제해결 기본］ 9일차 － 사칙연산 유효성 검사.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        for (int testcase = 1; testcase <= 10; testcase++) {
            int N = Integer.parseInt(br.readLine());
            boolean flag = true;

            for (int i = 0; i < N; i++) {
                String[] query = br.readLine().split(" ");
                if (query.length == 2) {
                    String value = query[1];
                    if (value.equals("+") || value.equals("-") || value.equals("*") || value.equals("/")) {
                        flag = false;
                    }
                } else {
                    String value = query[1];
                    if (value.equals("+") && value.equals("-") && value.equals("*") && value.equals("/")) {
                        flag = false;
                    }
                }
            }

            System.out.println("#" + testcase + " " + (flag ? 1 : 0));
        }
    }
}
